ruby/vampires.rb

```diff
@@ -1,10 +1,80 @@
-puts "What is your name?"
-vampire_name = gets.chomp
-puts "How old are you?"
-age = gets.chomp
-puts "What year were you born?"
-birth_year = gets.chomp
-puts "Our company cafeteria serves garlic bread. Should we order some for you?"
-garlic = gets.chomp
-puts "Would you like to enroll in the company's health insurance?"
-insurance = gets.chomp
\ No newline at end of file
+# Method to convert yes/no responses to booleans
+
+def boolean_convert(str)
+  if str == "yes"
+    str = true 
+  else 
+    str = false 
+  end
+end 
+
+# User interface
+
+puts "How many employees will be processed?"
+count = gets.chomp.to_i 
+
+x = 0
+
+# Loop processing code until through every employee
+while x < count
+
+  puts "What is your name?"
+  name = gets.chomp
+  
+  puts "How old are you?"
+  # Converting string response to integer 
+  age = gets.chomp .to_i
+  
+  puts "What year were you born?"
+  # Converting string response to integer 
+  year = gets.chomp.to_i 
+  
+  # Converting birth year to age
+  age_2 = 2017 - year
+  
+  # Check to see that the ages match
+  if age == age_2
+    age = true 
+  else 
+    age = false 
+  end 
+  
+  puts "Our company cafeteria serves garlic bread. Should we order some for you? (yes/no)"
+  garlic = boolean_convert(gets.chomp) 
+  
+  puts "Would you like to enroll in the company’s health insurance? (yes/no)"
+  insurance = boolean_convert(gets.chomp) 
+  
+  # Check for sunshine allergy
+  puts "Please list any allergies, type \"done\" when finished."
+  allergies = gets.chomp 
+  
+  until allergies == "done" || allergies == "sunshine"
+    puts "Please list any allergies, type \"done\" when finished."
+    allergies = gets.chomp
+  end
+  
+  if allergies == "sunshine"
+    puts "Probably a vampire"
+  else 
+    # Check to see the likelihood of being a vampire
+    if age == true && (garlic || insurance)
+      if name == "Drake Cula" || name == "Tu Fang"
```

Comment:
I how you are nesting conditions here.  Keep in mind that if the interviewee has the name Drac or Fang then they are definitely a vampire.

Considering this, it may be best to check this first.  If this comes up then there is no reason to run any of the other checks.